api/uniter/unit.go

```diff
@@ -947,6 +947,30 @@ func (b *CommitHookParamsBuilder) UpdateUnitState(state map[string]string) {
 	}
 }
 
+// AddStorage records a request for adding storage.
+func (b *CommitHookParamsBuilder) AddStorage(constraints map[string][]params.StorageConstraints) {
+	storageReqs := make([]params.StorageAddParams, 0, len(constraints))
+	for storage, cons := range constraints {
+		for _, one := range cons {
+			storageReqs = append(storageReqs, params.StorageAddParams{
+				UnitTag:     b.arg.Tag,
+				StorageName: storage,
+				Constraints: one,
+			})
+		}
+	}
+
+	b.arg.AddStorage = storageReqs
+}
+
+// SetPodSpec records a request to update the PodSpec for an application.
+func (b *CommitHookParamsBuilder) SetPodSpec(appTag names.ApplicationTag, spec string) {
+	b.arg.SetPodSpec = &params.EntityString{
```

Comment:
#11258 introduces new args for setting podspec on uniter v15
So there will need to be some tweaks to accommodate that.

Reply:
Thanks for the heads up! I will rebase and pull those changes in.